scripts/households.py

```diff
@@ -1,66 +1,65 @@
-    corresponding to b, and a S dim vector corresponding to n.
-
+    variable b, and the labor supply variable n
+    Note that we were dealing with three dimensional vectors earlier. We now
+    have an S-1 dim vector corresponding
+    to b, and an S dim vector corresponding to n.
 
     Refer to equations (4.9) and (4.10) of chapter 4 for details on where the
     variables foc_errors_b and foc_errors_n are coming from
 
 
     Args:
     b_sp1: The savings values for each period. The call to this function should
-           provide initial guesses for this variable.
-    n_s: The labor supply values for each period. The call to this function should
-         provide initial guesses for this variable.
+            provide initial guesses for this variable.
+    n_s: The labor supply values for each period. The call to this function
+            should provide initial guesses for this variable.
+    BQ_path: path of bequests
     l_tilde: maximum amount of labor supply
     chi: scale parameter
     theta: Frisch elasticity of labor supply
-    rho_s: risk that someone alive at age-s will die at the end of that period and not be alive for age s+1
-    n is not contained in the remaining arguments anymore. If someone decides to change this, please provide detailed documentation
+
+    n is not contained in the remaining arguments anymore. If someone decides
+    to change this, please provide detailed documentation
     on why you are doing so.
 
+
     Returns:
     foc_errors: A list where the first S-1 values are b2, b3, ..., bS, and
     the next S values are n1, n2, ..., nS.
     '''
 
-    beta, sigma, r, w, b_init, l_tilde, chi, theta, rho_s = args
-    b_s = np.append(b_init, b_sp1)  # When working on SS.py, note that b_sp1_guess is now of length S-1
-    b_sp1 = np.append(b_sp1, 0.0)
-    c = get_c(r, w, n_s, b_s, b_sp1)
-    mu_c = mu_cons(c, sigma)
-    mu_n = mu_labor(n_s, l_tilde, chi, theta)
-
-    # First get the Euler equations defined by Equation (4.10) - S-1 of these
-    lhs_euler_b = mu_c
-    rhs_euler_b = beta * (1+r) * mu_c
-    foc_errors_b = lhs_euler_b[:-1] - rhs_euler_b[1:]
-    # The above line doesn't need to be changed since it works for a general S as well (writing out the vectors helps to see this)
-
-    # Next get the Euler equations defined by (4.9) - S of these
-    lhs_euler_n = w * mu_c
-    rhs_euler_n = mu_n
-    foc_errors_n = lhs_euler_n - rhs_euler_n
+    BQpath, rho_s, beta, sigma, b_init, n, rpath, wpath = params
+    # rpath = length p, n = length p
+    # p = upsilonmber of periods left in life time
+    # wpath = length p, b_sp1 = length p-1
 
-    foc_errors = np.append(foc_errors_b, foc_errors_n)
+    b_s = np.append(b_init, b_sp1)
+    b_sp1 = np.append(b_sp1, 0.0)
 
-    return foc_errors
+    c = (1 + rpath) * b_s + wpath * n + BQpath - b_sp1
+    MU_c = mu_cons(c, sigma)
+    errors = (MU_c[:-1] - beta * (1 + rpath[1:]) *
```

Comment:
This is just the FOC for the choice of savings.  Why deleting the FOC for the choice of labor supply?